### ISSUE TITLE
Rename Group -> Team

### ISSUE THREAD

**Issue by craigfay** [zero-gravity-computer/issue-tracker-backend#5]
TLDR; `Group` should be replaced with `Organization` and `Team`.

`Groups` is probably an overly simplified representation of how people would want to organize `Issues` and `Contributors`.

Probably a more appropriate name for the `Group` and `GroupMembership` models is `Team` and `TeamMembership`, because the language is a closer fit to the actual use case.

We should also allow for multiple `Teams` per account. This would mean introducing an `Organization` model, which can contain multiple `Teams`. `Issues` would have a one-to-one relationship with both `Organization`, and `Team`, but the relationship with `Team` should be null-able.

This also means that we'd have a new `TeamMembership` model.

**Comment by craigfay** [comment]
@J4mesB4xter curious for your thoughts on if this would be a good feature or not.

**Comment by J4mesB4xter** [comment]
Seems smart. I'll get it underway.
